aiida/engine/daemon/execmanager.py

```diff
@@ -174,14 +175,18 @@ def upload_calculation(
     # Still, beware! The code file itself could be overwritten...
     # But I checked for this earlier.
     for code in input_codes:
-        if isinstance(code, PortableCode):
+        if isinstance(code, (PortableCode, PortableContainerizedCode)):
             # Note: this will possibly overwrite files
             for filename in code.base.repository.list_object_names():
                 # Note, once #2579 is implemented, use the `node.open` method instead of the named temporary file in
                 # combination with the new `Transport.put_object_from_filelike`
                 # Since the content of the node could potentially be binary, we read the raw bytes and pass them on
                 with NamedTemporaryFile(mode='wb+') as handle:
-                    handle.write(code.base.repository.get_object_content(filename, mode='rb'))
+                    try:
+                        handle.write(code.base.repository.get_object_content(filename, mode='rb'))
+                    except:
+                        # raise TypeError('directory not supperted.')
+                        pass
```

Comment:
Yes, I put the pass here for a further look into it and forget about it.  I think what I expected here is as you said to walk through the directory and copy all from inside. This will then also support the that `filepath_executable` of portable code is a real relative path of code inside a subfolder.